plugins/boltz/main.go

```diff
@@ -1,17 +1,112 @@
 package boltz
 
 import (
+	"fmt"
 	"time"
 
+	"github.com/BoltzExchange/boltz-lnd/boltz"
+	"github.com/bolt-observer/agent/entities"
 	agent_entities "github.com/bolt-observer/agent/entities"
+	"github.com/bolt-observer/agent/plugins"
+	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/golang/glog"
+	"github.com/tyler-smith/go-bip39"
+	"github.com/urfave/cli"
 )
 
+const (
+	DefaultBoltzUrl = "https://boltz.exchange/api"
+	SecretBitSize   = 256
+)
+
+var PluginFlags = []cli.Flag{
+	cli.StringFlag{
+		Name: "boltzurl", Value: DefaultBoltzUrl, Usage: "url of boltz api", Hidden: false,
+	},
+	cli.BoolFlag{
+		Name: "dumpmnemonic", Usage: "should we print master secret as mnemonic phrase (dangerous)", Hidden: false,
+	},
+	cli.StringFlag{
+		Name: "setmnemonic", Value: "", Usage: "update saved secret with this key material (dangerous)", Hidden: false,
+	},
+}
+
+func init() {
+	// Register ourselves with plugins
+	plugins.AllPluginFlags = append(plugins.AllPluginFlags, PluginFlags...)
+	plugins.RegisteredPlugins = append(plugins.RegisteredPlugins, plugins.PluginData{
+		Name: "boltz",
+		Init: func(lnAPI agent_entities.NewAPICall, cmdCtx *cli.Context) agent_entities.Plugin {
+			r := NewPlugin(lnAPI, cmdCtx)
+			return agent_entities.Plugin(r)
+		},
+	})
+}
+
 // Plugin can save its data here
 type Plugin struct {
+	BoltzAPI     *boltz.Boltz
+	ChainParams  *chaincfg.Params
+	LnAPI        entities.NewAPICall
+	MasterSecret []byte
+
 	agent_entities.Plugin
 }
 
+// NewPlugin creates new instance
+func NewPlugin(lnAPI agent_entities.NewAPICall, cmdCtx *cli.Context) *Plugin {
+	network := cmdCtx.String("network")
+
+	params := chaincfg.MainNetParams
+	switch network {
+	case "mainnet":
+		params = chaincfg.MainNetParams
+	case "testnet":
+		params = chaincfg.TestNet3Params
+	case "regtest":
+		params = chaincfg.RegressionNetParams
+	case "simnet":
+		params = chaincfg.SimNetParams
+	}
+
+	var (
+		entropy []byte
+		err     error
+	)
+
+	mnemonic := cmdCtx.String("setmnemonic")
+	// TODO: save this to db
+	if mnemonic != "" {
+		entropy, err = bip39.MnemonicToByteArray(mnemonic)
+		if err != nil {
+			glog.Fatalf("Invalid mnemonic: %v", err)
```

Comment:
Why do we have to fatal here. I suggest we propagate (return) errors and stop at the top if error is present.